webserver/server/app/controllers/DexMeanPriceController.ts

```diff
@@ -0,0 +1,116 @@
+import { Body, Controller, TsoaResponse, Res, Post, Route, SuccessResponse } from 'tsoa';
+import { StatusCodes } from 'http-status-codes';
+import { DEX_PRICE_LIMIT } from '../../../shared/constants';
+import tx from 'pg-tx';
+import pool from '../services/PgPoolSingleton';
+import { resolvePageStart, resolveUntilTransaction } from '../services/PaginationService';
+import type { ErrorShape } from '../../../shared/errors';
+import { genErrorMessage } from '../../../shared/errors';
+import { Errors } from '../../../shared/errors';
+import { expectType } from 'tsd';
+import type { EndpointTypes } from '../../../shared/routes';
+import { Routes } from '../../../shared/routes';
+import { getAddressTypes } from '../models/utils';
+import type { DexMeanPriceResponse } from '../../../shared/models/DexMeanPrice';
+import { dexMeanPrices } from '../services/DexMeanPrice';
+
+const route = Routes.dexMeanPrice;
+
+@Route('dex/mean-price')
```

Comment:
What is this query supposed to be for?

The use-case for this price feed is for wallets to have a webserver they can query to get the price of given assets

- Not clear why an address should be passed for this
- Not clear why a before/after tx should be passed for this
- Not clear why we're returning a list as a response for this

Can you clarify your vision for how a website would use this endpoint to show the latest price information about tokens?

The only thing I can think of is that this endpoint is meant as an intermediate step for some other price service that does the following:
1. Query the latest block
2. Query `dex/mean-price` for a list of known DEX addresses to find recent swaps (recent being defined as not too far from the best block queried in (1))
3. Aggregate these prices in the middleware according to some formula
4. Cache the aggregated result and return that
5. Invalidate the cache every X minutes and re-query from the database

Is this what you had in mind? If that's the case, I feel like using a materialized view is going to be significantly faster and more efficient than what we have here

Reply:
The idea for this endpoint was to enable querying historical (and current) data - e.g. if you want to draw a price chart for a given DEX, asset pair and a time interval. I agree that it's not well suited for getting the latest price and a separate endpoint would be more appropriate. Do you think this endpoint won't be useful? What is the preferred next step - adding the latest price endpoint?

Reply:
Okay, fair enough. So basically we want two endpoints then?